netbox/dcim/signals.py

```diff
@@ -180,3 +180,36 @@ def update_mac_address_interface(instance, created, raw, **kwargs):
     if created and not raw and instance.primary_mac_address:
         instance.primary_mac_address.assigned_object = instance
         instance.primary_mac_address.save()
+
+
+@receiver(post_save, sender=Location)
+@receiver(post_save, sender=Site)
+def sync_cached_scope_fields(sender, instance, **kwargs):
+    """
+    Rebuild cached scope fields for all CachedScopeMixin-based models
+    affected by a change in a Region, SiteGroup, Site, or Location.
+
+    This method is safe to run for objects created in the past and does
+    not rely on incremental updates. Cached fields are recomputed from
+    authoritative relationships.
+    """
+
```

Comment:
We can check whether the triggering instance is newly created, and if so skip it because we know it can't yet be referenced by other objects. (The function signature also needs to be updated to capture the `created` argument.)
```suggestion
    if created:
        return

```